auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,405 @@
+        )
+    )
+
+    @api.multi
+    def _check_db_exist(self):
+        for rec in self:
+            db_list = self.get_db_list(rec.host, rec.port)
+            if rec.name in db_list:
+                return True
+            return False
+
+    _constraints = [
+        (
+            _check_db_exist,
+            _('Error ! No such database exists!'), ['name']
+        )
+    ]
+
+    @api.multi
+    def test_sftp_connection(self):
+        confs = self.search([])
+        # Check if there is a success or fail and write messages
+        messageTitle = ""
+        messageContent = ""
+        for rec in confs:
+            # db_list = self.get_db_list(cr, uid, [], rec.host, rec.port)
+            try:
+                # pathToWriteTo = rec.sftppath
+                ipHost = rec.sftpip
+                portHost = rec.sftpport
+                usernameLogin = rec.sftpusername
+                passwordLogin = rec.sftppassword
+                # Connect with external server over SFTP, so we know sure that
+                # everything works.
+                srv = pysftp.Connection(host=ipHost, username=usernameLogin,
+                                        password=passwordLogin, port=portHost)
+                srv.close()
+                # We have a success.
+                messageTitle = _("Connection Test Succeeded!")
+                messageContent = _(
+                    "Everything seems properly set up for FTP back-ups!")
+            except Exception as e:
+                messageTitle = _("Connection Test Failed!")
+                if len(rec.sftpip) < 8:
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += "Here is what we got instead:\n"
+        if "Failed" in messageTitle:
+            raise except_orm(
+                _(messageTitle), _(
+                    messageContent + "%s") %
+                tools.ustr(e))
+        else:
+            raise Warning(_(messageTitle), _(messageContent))
+
+    @api.model
+    def schedule_backup(self):
+
+        for rec in self.search([]):
+            db_list = self.get_db_list(rec.host, rec.port)
+            if rec.name in db_list:
+                file_path = ''
+                bkp_file = ''
+                try:
+                    if not os.path.isdir(rec.bkp_dir):
+                        os.makedirs(rec.bkp_dir)
+                except:
+                    raise
+                # Create name for dumpfile.
+                bkp_file = '%s_%s.dump.zip' % (
+                    time.strftime('%d_%m_%Y_%H_%M_%S'),
+                    rec.name)
+                file_path = os.path.join(rec.bkp_dir, bkp_file)
+                conn = self.get_connection(rec.host, rec.port)
+                bkp = ''
+                try:
+                    bkp = execute(
+                        conn, 'dump', tools.config['admin_passwd'], rec.name)
```

Comment:
so you expect the remote admin password to be the same as the local admin password? this needs to be documented.